presto-druid/src/main/java/io/prestosql/plugin/druid/DruidJdbcClient.java

```diff
@@ -0,0 +1,228 @@
+    {
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            try (ResultSet resultSet = getTables(connection, schema, Optional.empty())) {
+                ImmutableList.Builder<SchemaTableName> list = ImmutableList.builder();
+                while (resultSet.next()) {
+                    list.add(new SchemaTableName(
+                            resultSet.getString("TABLE_SCHEM"),
+                            resultSet.getString("TABLE_NAME")));
+                }
+                return list.build();
+            }
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    public Optional<JdbcTableHandle> getTableHandle(JdbcIdentity identity, SchemaTableName schemaTableName)
+    {
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            String jdbcSchemaName = schemaTableName.getSchemaName();
+            String jdbcTableName = schemaTableName.getTableName();
+            try (ResultSet resultSet = getTables(connection, Optional.of(jdbcSchemaName), Optional.of(jdbcTableName))) {
+                List<JdbcTableHandle> tableHandles = new ArrayList<>();
+                while (resultSet.next()) {
+                    tableHandles.add(new JdbcTableHandle(
+                            schemaTableName,
+                            DRUID_CATALOG,
+                            resultSet.getString("TABLE_SCHEM"),
+                            resultSet.getString("TABLE_NAME")));
+                }
+                if (tableHandles.isEmpty()) {
+                    return Optional.empty();
+                }
+                if (tableHandles.size() > 1) {
+                    throw new PrestoException(NOT_SUPPORTED, "Multiple tables matched: " + schemaTableName);
+                }
+                return Optional.of(getOnlyElement(tableHandles));
+            }
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        switch (typeHandle.getJdbcType()) {
+            case Types.VARCHAR:
+                return jdbcTypeToPrestoType(typeHandle);
+            default:
+                return super.toPrestoType(session, connection, typeHandle);
+        }
+    }
+
+    private Optional<ColumnMapping> jdbcTypeToPrestoType(JdbcTypeHandle type)
+    {
+        int columnSize = type.getColumnSize();
+        if (columnSize > VarcharType.MAX_LENGTH || columnSize == -1) {
+            return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+        }
+        return Optional.of(varcharColumnMapping(createVarcharType(columnSize)));
+    }
+
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
```

Comment:
Why is this method overridden?

Reply:
This method was overriden since it was calling the getDruidColumns() method.